docs/Upgrade_Suggestions_Data_Analysis.md

```diff
@@ -0,0 +1,87 @@
+# Upgrade Suggestion (Data Analysis)
+
+Upgrading old Chef infrastructure could be very challenging, in former times,
+users would upload cookbooks and register Chef nodes (bootstrap) to a Chef
+Infra Server, such nodes would have a defined run-list, that is, the list of
+policies (cookbooks) applied to the node. The run-list could also be modified
+by roles and environments, giving the user a lot of flexibility, but with a
+very high level of complexity.
+
+As an example, let us think about a user with X number of nodes running Chef
+11, in order to upgrade these nodes to Chef 12 a series of things needs to
+happen:
+
+* Cookbooks need to be updated and tested to run on Chef 12
+* The Chef client version on the node(s) needs to be upgraded
+* (in some cases) A Chef Infra Server upgrade could be required
+
+This problem gets bigger as the number of nodes and cookbooks increases. There
+could be cookbooks that are being used by more than one node (a very common
+thing to do), and during an upgrade on a single node, without knowing, the user
+could break other nodes accidentally.
+
+Through the analysis of the Chef Infra Server data, we could give our users
```

Comment:
This is a good summary of what we're trying to accomplish -  I'd recommend moving it to be the introductory paragraph; with the background/need explained after.